testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveSparkCompatibility.java

```diff
@@ -423,6 +424,102 @@ private void testReadTrinoCreatedTable(String tableName, String tableFormat)
+        onSpark().executeQuery(format("INSERT INTO %s PARTITION(dt='2022-04-13 00:00:00') VALUES (2)", sparkTableName));
+        onSpark().executeQuery(format("INSERT INTO %s PARTITION(dt='2022-04-13 00:00') VALUES (3)", sparkTableName));
+        onSpark().executeQuery(format("INSERT INTO %s PARTITION(dt='12345-06-07') VALUES (4)", sparkTableName));
+        onSpark().executeQuery(format("INSERT INTO %s PARTITION(dt='123-04-05') VALUES (5)", sparkTableName));
+        onSpark().executeQuery(format("INSERT INTO %s PARTITION(dt='-0001-01-01') VALUES (6)", sparkTableName));
+
+        assertThat(onTrino().executeQuery("SELECT \"$partition\" FROM " + trinoTableName))
+                .containsOnly(List.of(
+                        row("dt=2022-04-13 00%3A00%3A00.000000000"),
+                        row("dt=2022-04-13 00%3A00%3A00"),
+                        row("dt=2022-04-13 00%3A00"),
+                        row("dt=12345-06-07"),
+                        row("dt=123-04-05"),
+                        row("dt=-0001-01-01")));
+
+        // Use date_format function to avoid exception due to java.sql.Date.valueOf() with 5 digit year
+        assertThat(onSpark().executeQuery("SELECT value, date_format(dt, 'yyyy-MM-dd') FROM " + sparkTableName))
+                .containsOnly(List.of(
+                        row(1, "2022-04-13"),
+                        row(2, "2022-04-13"),
+                        row(3, "2022-04-13"),
+                        row(4, "+12345-06-07"),
+                        row(5, null),
+                        row(6, "-0001-01-01")));
+
+        // Use date_format function to avoid exception due to java.sql.Date.valueOf() with 5 digit year
+        assertThat(onHive().executeQuery("SELECT value, date_format(dt, 'yyyy-MM-dd') FROM " + sparkTableName))
+                .containsOnly(List.of(
+                        row(1, "2022-04-13"),
+                        row(2, "2022-04-13"),
+                        row(3, "2022-04-13"),
+                        row(4, "12345-06-07"),
+                        row(5, "0123-04-06"),
+                        row(6, "0002-01-03")));
+
+        // Cast to varchar so that we can compare with Spark & Hive easily
+        assertThat(onTrino().executeQuery("SELECT value, CAST(dt AS VARCHAR) FROM " + trinoTableName))
+                .containsOnly(List.of(
+                        row(1, "2022-04-13"),
+                        row(2, "2022-04-13"),
+                        row(3, "2022-04-13"),
+                        row(4, "12345-06-07"),
+                        row(5, "0123-04-05"),
+                        row(6, "-0001-01-01")));
+
+        onTrino().executeQuery("DROP TABLE " + trinoTableName);
+    }
+
+    @Test(groups = {HIVE_SPARK, PROFILE_SPECIFIC_TESTS}, dataProvider = "unsupportedPartitionDates")
+    public void testReadSparkInvalidDatePartitionName(String inputDate, java.sql.Date outputDate)
```

Comment:
How much time does it take to execute the test (7 times)?

i guess it takes a while, so maybe we can create table once?
not as clean as your solution, but maybe more practical?

(that would be a follow up)

Reply:
It takes 18 sec in my laptop. The elapsed time looks acceptable to me. 
<img width="387" alt="Screen Shot 2022-04-13 at 18 25 46" src="https://user-images.githubusercontent.com/6237050/163145196-a5ec2e1a-0c66-443a-846c-d4b9dc272d09.png">